nni/common/graph_utils.py

```diff
@@ -282,27 +285,34 @@ def _expand_key_func_node(self, node, nodes, input_to_node, output_to_node,
         self.global_count += 1
         op_type = node.kind()
         node_group = [node]
-        inputs = list()
-        outputs = list()
+        inputs = set()
+        outputs = set()
         node_queue = queue.Queue()
         node_queue.put(node)
         while not node_queue.empty():
             curr_node = node_queue.get()
             for _input in curr_node.inputs():
+                if _input.node().kind() == CONSTANT_KIND:
+                    continue
                 input_name = _input.debugName()
-                if input_name in output_to_node and output_to_node[input_name] in nodes:
-                    predecessor_node = output_to_node[input_name]
-                    if not self._is_key_func(predecessor_node):
-                        node_group.append(predecessor_node)
-                        node_queue.put(predecessor_node)
-                    else:
-                        inputs.append(input_name)
+                if input_name in output_to_node:
+                    for predecessor_node in output_to_node[input_name]:
+                        if predecessor_node in nodes:
+                            if not self._is_key_func(predecessor_node):
+                                node_group.append(predecessor_node)
+                                node_queue.put(predecessor_node)
+                            else:
+                                inputs.add(input_name)
```

Comment:
If predecessor_node is already merged into this node group, then I guess it's output should not be taken as the input of this node group?

Reply:
checking added, please review again.

Reply:
I guess I didn't explain myself clearly, sorry about that. 
What I want to say is that maybe we can remove the `else` part of the following code.
```
 if not self._is_key_func(predecessor_node):
    node_group.append(predecessor_node)
    node_queue.put(predecessor_node)
 else:
    inputs.add(input_name)
```

Because whether the `input_name` should be added to the inputs(`inputs`) of this nodes group is only decided by if `predecessor_node` is in `nodes`? It has nothing to do with whether `predecessor_node` is a key node/func? I guess? Please correct me if I'm wrong. Thanks~

Reply:
It does not work removing the `else` as I tested. Because if the predecessor node is key func, the predecessor node will be merged into another node.  There is only one key func node within one merged node.

Reply:
Got it! Thanks~